server/src/main/java/org/opensearch/cluster/decommission/DecommissionService.java

```diff
@@ -303,19 +309,66 @@ public void onFailure(Exception e) {
         }
     }
 
-    private void failDecommissionedNodes(ClusterState state) {
+    private void weighAwayDecommissionedZone(ClusterState state, TimeValue timeOutForNodeDraining) {
         // this method ensures no matter what, we always exit from this function after clearing the voting config exclusion
         DecommissionAttributeMetadata decommissionAttributeMetadata = state.metadata().decommissionAttributeMetadata();
         DecommissionAttribute decommissionAttribute = decommissionAttributeMetadata.decommissionAttribute();
+
+        // Awareness values refers to all zones in the cluster
+        List<String> awarenessValues = forcedAwarenessAttributes.get(decommissionAttribute.attributeName());
+
+        decommissionController.updateMetadataWithDecommissionStatus(DecommissionStatus.WEIGH_AWAY, new ActionListener<>() {
+            @Override
+            public void onResponse(DecommissionStatus status) {
+                logger.info("updated the decommission status to [{}]", status);
+                // set the weights
+
+                decommissionController.setWeightForDecommissionedZone(awarenessValues, new ActionListener<>() {
+                    @Override
+                    public void onResponse(ClusterPutWRRWeightsResponse response) {
```

Comment:
Does this response just sets the weight or also ensure node is drained?

Reply:
It just set the weight. We are adding timeout for nodes to be drained and log the active connection on the nodes being decommissioned.